prow/plank/controller_test.go

```diff
@@ -1361,3 +1361,73 @@ func TestMaxConcurrencyWithNewlyTriggeredJobs(t *testing.T) {
 		}
 	}
 }
+
+func TestMaxConcurency(t *testing.T) {
```

Comment:
nit: can we get a test case where:

 -  it actually exceeds and does not equal the max concurrency
 - we have other jobs but they're newer, not older, so we _can_ execute
 - we have other jobs and they're older but not pending so we _can_ execute